pkg/apis/gcp/validation/infrastructure.go

```diff
@@ -84,6 +84,10 @@ func ValidateInfrastructureConfig(infra *apisgcp.InfrastructureConfig, nodesCIDR
 		allErrs = append(allErrs, field.Invalid(networksPath.Child("vpc", "cloudRouter"), infra.Networks.VPC.CloudRouter, "cloud router can not be configured when the VPC name is not specified"))
 	}
 
+	if infra.Networks.CloudNAT != nil && infra.Networks.CloudNAT.NatIPs != nil {
```

Comment:
What's the reason for this limitation? Let's say I want to attach two NAT IPs, one by name and one by IP. Why is this forbidden/not possible?

Reply:
The data source requires a unique name and we need it to extract the ip address for the status and get the selflink for the nat gateway. If there is a better way I'm happy to adapt it.

Reply:
I don't understand. You say this `<NAME>` here must be unique?

```terraform
data "google_compute_address" "<NAME>" {
```

What's the issue with allowing IPs then?

Reply:
```terraform
data "google_compute_address" "Name" {
  name = "<NAME>"
}
```

The ```<NAME>``` here must be unique, how will I get the name for the ip? We could implement it via the sdk but not sure, if that will suit the use case.

Reply:
OK, I see, understood, but why do we have then https://github.com/gardener/gardener-extension-provider-gcp/pull/39/files#diff-50ba7c5f1771b36c61263800bc8ea823R122 in the first place?

Reply:
For convenience, when you want to debug something or look them up, you don't have to go each and every time to the google cloud console. If you think it's better to remove it from the status we can also do that. But I think it will make things easier for us.

Reply:
It's fine to have it in the status, but as of now an end-user can configure `.networks.cloudNAT.natIPs` with should not be possible as it cannot be used/nothing is implemented.

Reply:
https://github.com/gardener/gardener-extension-provider-gcp/pull/39/files#diff-4863b24e73b205b0206ea0c2b7e55d86R88 isn't this doing the validation?

Reply:
Or do you mean something else?

Reply:
My point is that it shouldn't be in the API in the first place if it's not supported.

Reply:
Got it, so what do you suggest? Removing it from CloudNat and adding it only to the status or removing it at all?

Reply:
I‘m suggesting to remove it from the spec. If you consider it helpful having it in the status then I‘m fine with it.